cmp/store/impl/StoreFilterFieldImplModel.js

```diff
@@ -115,38 +117,41 @@ export class StoreFilterFieldImplModel {
     // Implementation
     //------------------------
     applyFilter() {
-        this.store?.setFilter(this.filter);
+        this.store?.filterModel.replaceFilters(this.filter);
     }
 
     regenerateFilter() {
-        const {filter, filterText, filterOptions} = this,
+        const {filter, filterText} = this,
             activeFields = this.getActiveFields(),
             supportDotSeparated = !!activeFields.find(it => it.includes('.')),
             searchTerm = escapeRegExp(filterText),
             initializing = isUndefined(filter);
 
-        let fn = null;
-        if (searchTerm && !isEmpty(activeFields)) {
+        let {testFn} = this;
+        if (!testFn && searchTerm && !isEmpty(activeFields)) {
             const regex = new RegExp(`(^|\\W)${searchTerm}`, 'i');
-            fn = (rec) => activeFields.some(f => {
+            testFn = (rec) => activeFields.some(f => {
                 // Use of lodash get() slower than direct access - use only when needed to support
                 // dot-separated field paths. (See note in getActiveFields() below.)
                 const fieldVal = supportDotSeparated ? get(rec.data, f) : rec.data[f];
                 return regex.test(fieldVal);
             });
         }
 
-        const newFilter = fn ? {...filterOptions, fn} : null;
-        if (filter === newFilter) return;
-
+        const newFilter = testFn ? new FunctionFilter({id: this.xhId, testFn}) : null;
+        if (!newFilter && this.filter) {
+            this.store?.filterModel.removeFilters(this.filter);
+        }
         this.filter = newFilter;
         if (!initializing && this.onFilterChange) this.onFilterChange(newFilter);
 
         // Only respect the buffer for non-null changes.  Allows immediate initialization and quick clearing
         if (!initializing && newFilter) {
             this.bufferedApplyFilter();
         } else {
-            this.applyFilter();
+            // start() here required to resolve a "Cannot update during an existing state transition"
+            // React error. Further investigation required.
+            start(() => this.applyFilter());
```

Comment:
Thanks for the note here - don't have any immediate answer and did not look at this in any detailed way, but calling out via PR comment so we can track and resolve.